### DISCUSSION TITLE
Coupling Questions

### DISCUSSION THREAD

**Discussion by notstarboard** [StormSurgeLive/asgs#1042] [Q&A] [answered]
Hi folks! From looking around the GitHub, it looks like ASGS supports ADCIRC-SWAN coupling. Have you considered supporting / would you have any interest in supporting ADCIRC-WAVEWATCH3 coupling as well? That's something we'd be interested in.

As for the ADCIRC-SWAN coupling, is it as simple as setting "WAVES=on" in the config file or are there other steps to get that working? Does this answer change if we're using a custom version of ADCIRC? Thanks!

**Comment by wwlwpd** [comment]
@jasonfleming will have to review this answer about SWAN coupling; but I believe that for basic support `WAVES=on` should do it. However, you will need to make sure you have `padcswan` and `unhcat.exe` also available. If ADCIRC was build via `build adcirc` then the you should be all set.

For WW3, @jasonfleming will have to answer. SWAN coupling is achieved via `adcswan` and `padcswan`, can you provide some more information (for me) about how model coupling is achieved with WW3? Is it file based coupling or are there other kinds of methods available?

It wouldn't hurt to create a _feature request_ issue in the tracker to "support WW3 coupling", as it would kick off any necessary requirements gathering towards that end.

Cheers!

**Comment by jasonfleming** [comment, marked as answer]
Yes @notstarboard you can activate SWAN coupling just by setting `WAVES=on` in your ASGS configuration file. You don't need to restart ASGS; the change will be picked up automatically on the next nowcast cycle. If your custom version of ADCIRC uses a different version of SWAN, and this requires changes to the `fort.26` template file or the `swaninit` file, then those files would be needed as well. 

Yes, we're interested in supporting WW3 coupling! :-) As @wwlwpd said, this would require upgrades to the build subsystem for the ADCIRC WW3 coupled model. We would also need to look at requirements for input files, any workflow associated with `adcprep`, specifying/handling output, etc. I agree with the feature request issue creation @wwlwpd mentioned above. Once that is created, should we set up Zoom call to discuss?

**Comment by notstarboard** [comment]
Thanks @wwlwpd & @jasonfleming ! That's all good to know and it's great that the SWAN coupling is so painless. I'd be more than happy to join a call to discuss, although I may also also want some reinforcements depending on what you're hoping to accomplish! Isaac recently mentioned wanting to run ADCRIC coupled (ideally with WW3) for nor'easters, so that's where I'm coming from - I figured I'd look into what's possible using ASGS. 

If you'd like to discuss the coupling itself and details thereabout, it might make sense to have Angelos on too. I don't know any of the specifics, but I know a new version came out somewhat recently that Angelos has been testing with. He may be able to provide some more details, and if not we might need some NOAA folks to chime in.

If you're curious about our interests in order to determine priority for this and such, it might be worth having Isaac on too. I don't know exactly how strongly Isaac feels about coupling via WW3 (as opposed to, say, SWAN) or running via ASGS vs handling things manually; I just know that running ADCIRC-WW3 in the most straightforward way is the goal. From my perspective running via ASGS would be a great solution if possible!

I also don't know how funding typically works for your work - where my mind's going: at my old job we had this "change order" concept, which was basically where customers could pay in order to fast-track enhancements that we couldn't otherwise prioritize. If you guys usually use a similar mechanism and/or if there's anything funding related that'd be tied up in this, I'd rather have Isaac speak to that :)

I'll create the issue for WW3 coupling support now, though, and we can go from there. Happy Thanksgiving to you both!

**Reply by jasonfleming** [reply]
Thank you @notstarboard I will follow up with you and @wwlwpd on specifics via email. :-)
